### ISSUE TITLE
Keynote manager gets locked with orphaned .lock file

### ISSUE THREAD

**Issue by JLee415** [eirannejad/pyRevit#1235]
**Describe the bug**
A user of ours are getting the "Wait for unlock timeout" error on the Keynote Manager and everybody else on the team is unable to run the Keynote Manager either. Currently our keynote .txt file is placed on BIM 360, and per the closed Issue #967 I have deleted the ".lock" file but we are still getting the same "Wait for Unlock timeout" error. 

![image](https://user-images.githubusercontent.com/83681718/117088259-c915f300-ad06-11eb-9b1a-5d47f43fbc19.png)

I have asked the user to uninstall and reinstall PyRevit to see if it fixes or clears this error but no luck either. I also asked the user to use another machine, download PyRevit and then run the Keynote Manager and he was able to get into Keynote with no issues on the other machine. But he noticed that on one of the notes he was editing/adding previously it said that "UserName is editing" and he is unable to delete or do anything with that note.

![image](https://user-images.githubusercontent.com/83681718/117088442-4b061c00-ad07-11eb-9a80-de8a6ad9047d.png)

**To Reproduce**
1. Click on the Keynote Manager Tool on the PyRevit tab.

**Expected behavior**
The Keynote Manager to load

**Screenshots**
See images above

**Desktop (please complete the following information):**
 - OS: Windows
 - pyRevit Version: 4.8.7 & 4.8.4
 - pyRevit Environment: Open a command prompt and run the command below. Replace these lines with the results. Take a look at this markdown guide and wrap the command results in ``` when pasting here for correct formatting.

`==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive/image)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.8.7" | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
==> Attachments
master | Product: "2021 First Customer Ship" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2020.2.3" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2019.2.2 Hotfix" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2018.3.3 Security Fix" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2017.2.5 Security Fix" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2016.1.9 Security Fix" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2015 Release 2 Update Release 13 (Subscription only release)" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
master | Product: "2014 Update Release 3" | Engine: IPY277 (277) | Path: "C:\Users\jun.lautan\AppData\Roaming\pyRevit-Master"
==> Installed Extensions
==> Default Extension Search Path
C:\Users\jun.lautan\AppData\Roaming\pyRevit\Extensions
==> Extension Search Paths
==> Extension Sources - Default
https://github.com/eirannejad/pyRevit/raw/master/extensions/extensions.json
==> Extension Sources - Additional
==> Installed Revits
2021 First Customer Ship | Version: 21.0.0.383 | Build: 20200220_1100(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2021\"
2020.2.3 | Version: 20.2.30.42 | Build: 20200826_1250(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2020\"
2019.2.2 Hotfix | Version: 19.2.20.24 | Build: 20190808_0900(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2019\"
2018.3.3 Security Fix | Version: 18.3.3.18 | Build: 20190510_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2018\"
2017.2.5 Security Fix | Version: 17.0.1169.0 | Build: 20190508_0315(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2017\"
2016.1.9 Security Fix | Version: 16.0.1218.0 | Build: 20190508_0715(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2016\"
2015 Release 2 Update Release 13 (Subscription only release) | Version: 15.0.1243.0 | Build: 20160220_0715(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2015\"
2014 Update Release 3 | Version: 13.11.4 | Build: 20140709_2115(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2014\"
==> Running Revit Instances
==> User Environment
Microsoft Windows 10 [Version 10.0.18363]
Executing User: GLOBAL\jun.lautan
Active User: GLOBAL\jun.lautan
Admin Access: No
%APPDATA%: "C:\Users\jun.lautan\AppData\Roaming"
Latest Installed .Net Framework: 4.8
No .Net Target Packs are installed.
No .Ne-Core Target Packs are installed.
pyRevit CLI 4.8.7.0`

**Comment by eirannejad** [comment]
See this please https://github.com/eirannejad/pyRevit/issues/967#issuecomment-834997605

**Comment by eirannejad** [comment]
So the `.lock` file is only created when working with locally shared keynote files. Here is some info that might help debugging the issue on your side.

- When a user opens the keynote manager (KM), it locks the keynote file on the BIM360 using the Autodesk Desktop Connector (ADC) API
- When a user edits a keynote, KM will write some info inside the keynote file (hidden to Revit) that a user has borrowed that specific keynote for editing. If you have a sticky lock on a specific keynote, manually clear the content of `Locks Table` in the keynote file. These locks are written in between the two lines below
```
# --------------------- @table(locks_table:"Locks Table")
>>> locks are listed here...
# ===================== @db(keynotesdb:"pyRevit Keynotes Manager DB")
```

For example a lock with id `tf5n6dURvk+JrZQ5r3gD5g==` exists in this keynote file example:

```
# @datastore("txt") @source("C:\Users\ehsan\Desktop\Keynotes.txt") @version("0.1") @encoding("UTF-16")
# ===================== @db(txn_db:"Internal DB")
# --------------------- @table(locks_table:"Locks Table")
# tf5n6dURvk+JrZQ5r3gD5g==	leo-wx-rir	eirannejad	w8/8rO/UOUyP0vs1oS53zA==	keynotesdb	keynotes	K7
# ===================== @db(keynotesdb:"pyRevit Keynotes Manager DB")
C1	CATEGORY ONE
C2	OTHER CATEGORY
# --------------------- @table(keynotes:"Keynotes Table")
```

So to clear the locks, remove the lock lines. Coordinate this with your team so you are the only person changing the file

```
# @datastore("txt") @source("C:\Users\ehsan\Desktop\Keynotes.txt") @version("0.1") @encoding("UTF-16")
# ===================== @db(txn_db:"Internal DB")
# --------------------- @table(locks_table:"Locks Table")
# ===================== @db(keynotesdb:"pyRevit Keynotes Manager DB")
C1	CATEGORY ONE
C2	OTHER CATEGORY
# --------------------- @table(keynotes:"Keynotes Table")
```

**Comment by Architectuur** [comment]
@eirannejad I have just run into this issue. I had a lock in the keynote file, I removed the lock as directed. I was able to get in once but then tried to get in again and the same message came up. I went into the keynote file to remove the lock, but there are no locks... Is there anything I can send you that would help?

**Comment by Architectuur** [comment]
@eirannejad Is there any update to this? I just downloaded the latest 4.8.10 and still no luck. This has crippled pyrevit keynotes for me. Is there anything I can supply to help fix this? The keynote text file is completely clean and I still can't even open the pyrevit keynote manager. I have attached the text file.

[Project Keynotes.txt](https://github.com/eirannejad/pyRevit/files/8035371/Project.Keynotes.txt)

**Comment by eirannejad** [comment]
What do you mean still no luck? I just tested 4.8.10 and it works 🥺 Would you please make sure you are running this? https://github.com/eirannejad/pyRevit/releases/tag/v4.8.10.22040%2B1743

**Comment by Architectuur** [comment]
I really know nothing about python or how github pulls the compiled code, but this is the error:
![image](https://user-images.githubusercontent.com/69218770/153274804-9d743195-ac8c-4226-8e75-9ad8cf89b430.png)
I downloaded 4.8.10 this morning. However the error is referencing 4.8.1.0.

**Comment by eirannejad** [comment]
Do you have a hidden `.lock` file where `Project Keynotes.txt` is?
The 4.8.1.0 is really 4.8.10. I'm fixing that #1481

**Comment by Architectuur** [comment]
Dang, I'm sorry. I have seen all of this talk about .lock files and totally misunderstood them and where they would be located. There was a .lock file. I deleted it and we are working now. I wish I would figured that out months ago. Many thanks to you.

**Comment by jlholt09** [comment]
I am using BIM 360 and downloaded the latest pyrevit 4.8.10 and I am still getting the error. I downloaded the .txt file from BIM 360, deleted the locked line and then reuploaded the file to BIM 360. And I still got the error. I've attached the keynote file.

![image](https://user-images.githubusercontent.com/100320679/155425894-5b41d035-9b8c-476c-8901-bb3083a453cf.png)

[Hunter Park - Keynotes.txt](https://github.com/eirannejad/pyRevit/files/8128447/Hunter.Park.-.Keynotes.txt)
